### ISSUE TITLE
Add Cache task

### ISSUE THREAD

**Issue by anna-geller** [kestra-io/kestra#1245]
## Problem

Imagine the following use case: you first extract data from 3 different sources. Subsequent tasks combine this data together before loading it to the final destination. The problem is that one of those 3 data sources is large, and it only needs to be extracted once per day despite the hourly schedule of the flow. 

## Feature

Add a `Cache` task. 

Benefits: allows to execute e.g. that daily task only once per day despite a regular hourly schedule

New state : `CACHED` to indicate users that the related tasks have been cached (in the execution views)

## Implementation
Below are two options to consider:

### Dedicated parent task (preferable)

Pros: 
- can cache the execution of an entire group of tasks at the same time if needed
- DAG topology view shows which tasks are going to be cached even before runtime 

```yaml
  - id: cache 
    type: io.kestra.core.tasks.cache.Cache 
    duration: PT5M
    tasks: 
      - id: download
        type: io.kestra.plugin.fs.http.Download
        uri: "https://www.data.gouv.fr/fr/datasets/r/d33eabc9-e2fd-4787-83e5-a5fcfb5af66d"
```

### Property on a core task

Pros: 
- granularity
- no extra task is needed

Cons:
- requires change on the core component
- some boilerplate if needs to be applied to multiple tasks

```yaml
        id: download
        type: io.kestra.plugin.fs.http.Download
        uri: "https://www.data.gouv.fr/fr/datasets/r/d33eabc9-e2fd-4787-83e5-a5fcfb5af66d"
        cache: 
          duration: PT5M
```

**Comment by Ben8t** [comment]
Note: it's caching for ouputs/state not file system

**Comment by anna-geller** [comment]
you mean not Luigi-style targets @Ben8t? 

this is mostly about caching the task state so if a task is cached and it generated some results that must be passed downstream, then that result must also be cached

**Comment by Ben8t** [comment]
Yes, we opened another issue for file system (Ludo/Loic agreed it's closer to the workingdir feature atm) #1275

**Comment by anna-geller** [comment]
closing as this was open for too long and apparently no demand for now from users -- we'll reopen when needed
